extensions/notebook/src/book/bookTocManager.ts

```diff
@@ -206,6 +208,11 @@ export class BookTocManager implements IBookTocManager {
 		}
 		if (isModified) {
 			await fs.writeFile(tocPath, yaml.safeDump(this.tableofContents, { lineWidth: Infinity, noRefs: true, skipInvalid: true }));
+			let toc: UndoRedoState = {
```

Comment:
Why are you doing this here? Shouldn't you be getting the state when you're actually adding the undo/redo item instead? The manager shouldn't have to be storing anything for the undo/redo stuff, that should generally all be handled by the service (and items you push to it)